akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
-          .stimulate(_ ⇒ ctx.schedule(1.second, self, Pong2), _ ⇒ Pong2)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong1)
-
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (ctx, Inert) ⇒
+          ctx.schedule(1.second, probe.ref, InertEvent)
+          Behaviors.same
+        case (_, Ping) ⇒
+          probe.ref ! Pong
+          Behaviors.same
+        case (_, ReceiveTimeout) ⇒
+          probe.ref ! GotReceiveTimeout
+          Behaviors.same
+        case (ctx, SetTimeout(duration)) ⇒
+          ctx.setReceiveTimeout(duration, ReceiveTimeout)
+          probe.ref ! TimeoutSet
+          Behaviors.same
+      }.decorate)
+      actor ! SetTimeout(1.minute)
+      probe.expectMessage(TimeoutSet)
+      actor ! Inert
+      probe.expectMessage(InertEvent)
+      actor ! Ping
+      probe.expectMessage(Pong)
     }
 
     "schedule a message" in {
-      sync(setup("ctx32") { (ctx, startWith) ⇒
-        startWith(_ ! Schedule(1.nano, ctx.self, Pong2, ctx.self))
-          .expectMultipleMessages(expectTimeout, 2) { (msgs, _) ⇒
-            msgs should ===(Scheduled :: Pong2 :: Nil)
-          }
-      })
-    }
-
-    "create a working adapter" in {
-      sync(setup("ctx40", ignorePostStop = false) { (ctx, startWith) ⇒
-        startWith.keep { subj ⇒
-          subj ! GetAdapter(ctx.self)
-        }.expectMessage(expectTimeout) { (msg, subj) ⇒
-          val Adapter(adapter) = msg
-          ctx.watch(adapter)
-          adapter ! Ping(ctx.self)
-          (subj, adapter)
-        }.expectMessage(expectTimeout) {
-          case (msg, (subj, adapter)) ⇒
-            msg should ===(Pong1)
-            ctx.stop(subj)
-            adapter
-        }.expectMulti(expectTimeout, 2) { (msgs, adapter) ⇒
-          msgs.toSet should ===(Set(Left(Terminated(adapter)(null)), Right(GotSignal(PostStop))))
-        }
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (ctx, Ping) ⇒
+          ctx.schedule(1.nano, probe.ref, Pong)
+          Behaviors.same
+      }.decorate)
+      actor ! Ping
+      probe.expectMessage(Pong)
     }
 
     "create a named adapter" in {
-      sync(setup("ctx41") { (ctx, startWith) ⇒
-        startWith.keep { subj ⇒
-          subj ! GetAdapter(ctx.self, "named")
-        }.expectMessage(expectTimeout) { (msg, subj) ⇒
-          val Adapter(adapter) = msg
-          adapter.path.name should include("named")
-        }
-      })
+      val probe = TestProbe[ActorRef[String]]()
+      val actor = spawn(Behaviors.immutablePartial[String] {
+        case (ctx, name) ⇒
+          probe.ref ! ctx.spawnMessageAdapter(identity, name)
+          Behaviors.same
+      }.decorate)
+      val adapterName = "hello"
+      actor ! adapterName
+      val adapter = probe.expectMessageType[ActorRef[String]]
+      adapter.path.name should include(adapterName)
```

Comment:
If we aren't reintroducing the omitted adapter test above, this should also actually use the adapter and see that it adapts and passes on the message to the actor.